lib/src/rules/prefer_final_fields.dart

```diff
@@ -64,6 +64,27 @@ class GoodMutable {
 ```
 ''';
 
+bool _isMutated(
+        VariableDeclaration variable, CompilationUnit compilationUnit) =>
+    DartTypeUtilities
+        .traverseNodesInDFS(compilationUnit)
+        .where((n) =>
+            n is SimpleIdentifier && n.bestElement is PropertyAccessorElement)
+        .any((n) {
+      SimpleIdentifier identifier = n as SimpleIdentifier;
+      PropertyAccessorElement bestElement =
+          identifier.bestElement as PropertyAccessorElement;
+      if (bestElement.variable != variable.element) {
+        return false;
+      }
+
+      return identifier.getAncestor((a) =>
+              a is AssignmentExpression ||
+              a is PrefixExpression ||
+              a is PostfixExpression) !=
+          null;
```

Comment:
Assuming this was in response to my comment on line 82 (in case that assumption is wrong).

The point was to prevent false negatives. Just because a field is referenced somewhere inside an assignment expression doesn't mean that it is being modified.